src/main/java/org/citopt/connde/web/rest/RestComponentFilterController.java

```diff
@@ -37,9 +38,37 @@
     @Autowired
     private RuleActionRepository ruleActionRepository;
 
+    @Autowired
+    private RuleTriggerRepository ruleTriggerRepository;
+
     @Autowired
     private RuleRepository ruleRepository;
 
+    /**
+     * Returns a list of rules that make use of a certain rule trigger.
+     *
+     * @param ruleTriggerId The id of the rule trigger for which using rules should be found
+     * @return A list of all rules that make use of the rule trigger
+     */
+    @GetMapping("/rules/by-ruleTrigger/{id}")
+    @ApiOperation(value = "Retrieves the rules which make use of a certain rule trigger and for which the user is authorized", produces = "application/hal+json")
+    @ApiResponses({@ApiResponse(code = 200, message = "Success")})
+    public ResponseEntity<List<Rule>> getRulesByRuleTriggerID(@PathVariable(value = "id") @ApiParam(value = "ID of the rule trigger", example = "5c97dc2583aeb6078c5ab672", required = true) String ruleTriggerId) {
+
+        List<Rule> rules = ruleRepository.findAll();

```

Comment:
Would it be possible to make use of `RuleTrigger ruleTrigger = ruleTriggerRepository.findOne(ruleTriggerId);` here? This would ensure that the deleting user really has access to the rule trigger for which the ID has been provided.

Reply:
Hmm I understand. I think I assumed the user would always have access to it as he/she is the one who is deleting. But I agree that it is good to check so that the function can be more secure and can be used in other contexts!

Reply:
Yes, you are right, it's not a (big) security risk. We just should avoid that a user is able to check what rules/triggers a foreign user may have defined or which rules are shared with this user. :-)